onnxruntime/core/providers/rocm/nn/dropout_impl.cu

```diff
@@ -0,0 +1,105 @@
+#include "hip/hip_runtime.h"
+/**
+* Copyright (c) 2016-present, Facebook, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* Modifications Copyright (c) Microsoft. */
+
+#include "core/providers/rocm/cu_inc/common.cuh"
+#include "core/providers/rocm/nn/dropout_impl.h"
+#include <hiprand_kernel.h>
+#include <algorithm>
+
+namespace onnxruntime {
+namespace rocm {
+
+constexpr int UNROLL = 4;
+
+template <typename T>
+__global__ void DropoutKernel(
+    const int64_t N,
+    const float ratio,
+    const std::pair<uint64_t, uint64_t> seeds,
+    const T* X_data,
+    T* Y_data,
+    bool* mask_data) {
+  const float p = 1.0f - ratio;
+  const float scale = 1.0f / p;
+
+  HIP_LONG idx = blockDim.x * blockIdx.x + threadIdx.x;
+  HIP_LONG step_size = gridDim.x * blockDim.x * UNROLL;
+  HIP_LONG rounded_size = ((N - 1) / step_size + 1) * step_size;
+
+  hiprandStatePhilox4_32_10_t state;
+  hiprand_init(seeds.first, idx, seeds.second, &state);
+
+  // We ensure every thread generates the same number of random numbers (by rounding
+  // up the size) and at the same timestep (by syncing threads).
+  // From HIP hiprand documentation:
```

Comment:
Maybe leave it from "CUDA curand ..", not sure.

Reply:
Yeah, it is from CUDA. Will delete it to avoid confusion.